app/src/main/java/com/oracle/javafx/scenebuilder/app/SceneBuilderApp.java

```diff
@@ -412,31 +412,19 @@ public void handleLaunch(List<String> files) {
         sendTrackingStartupInfo();
 
         if (showWelcomeDialog) {
-            // Creates an empty document
-            final DocumentWindowController newWindow = makeNewWindow();
-            newWindow.updateWithDefaultContent();
-            newWindow.openWindow();
-
             // Show ScenicView Tool when the JVM is started with option -Dscenic.
             // NetBeans: set it on [VM Options] line in [Run] category of project's Properties.
             if (System.getProperty("scenic") != null) { //NOI18N
-                Platform.runLater(new ScenicViewStarter(newWindow.getScene()));
+                // commented out until it is implemented
+                //Platform.runLater(new ScenicViewStarter(newWindow.getScene()));
             }
 
-            WelcomeDialogWindowController.getInstance().getStage().setOnHidden(event -> {
-                showUpdateDialogIfRequired(newWindow, () -> {
-                    if (!Platform.isFxApplicationThread()) {
-                        Platform.runLater(() -> showRegistrationDialogIfRequired(newWindow));
-                    } else {
-                        showRegistrationDialogIfRequired(newWindow);
-                    }
-                });
-            });
-
             // Unless we're on a Mac we're starting SB directly (fresh start)
             // so we're not opening any file and as such we should show the Welcome Dialog
             WelcomeDialogWindowController.getInstance().getStage().show();
 
+            // let JavaFX handle above call ASAP and delay empty document window for improved UX
+            makeEmptyDocumentWindowOnNextTick();
```

Comment:
The method name is confusing (to me). 

Can we rename it to `createEmptyDocumentWindowOnNextPulse` ?

Reply:
sure, done.